packages/http-server/src/getHttpConfigFromRequest.ts

```diff
@@ -1,26 +1,37 @@
-import { IHttpOperationConfig, IHttpRequest } from '@stoplight/prism-http';
+import { IHttpOperationConfig, IHttpRequest, ProblemJsonError, UNPROCESSABLE_ENTITY } from '@stoplight/prism-http';
+import { pipe } from 'fp-ts/lib/pipeable';
+import * as E from 'fp-ts/lib/Either';
+import * as t from 'io-ts';
+import { failure } from 'io-ts/lib/PathReporter';
+import { BooleanFromString } from 'io-ts-types/lib/BooleanFromString';
+//@ts-ignore
+import * as parsePreferHeader from 'parse-prefer-header';
 
-export const getHttpConfigFromRequest = (req: IHttpRequest): Partial<IHttpOperationConfig> => {
-  const httpOperationConfig: Partial<IHttpOperationConfig> = {};
-  const query = req.url.query;
+const PreferencesDecoder = t.union([
+  t.undefined,
+  t.partial(
+    {
+      code: t.string,
+      dynamic: t.string.pipe(BooleanFromString),
+      example: t.string,
+    },
+    'Preferences'
+  ),
+]);
 
-  if (!query) {
-    return {};
-  }
+type requestPreference = Partial<Omit<IHttpOperationConfig, 'mediaType'>>;
```

Comment:
```suggestion
type RequestPreference = Partial<Omit<IHttpOperationConfig, 'mediaType'>>;
```

Lower-case type name may mix with the variable name. At least during review on GitHub.